### PR TITLE
DSD-1021 - Heading case styles

### PR DESCRIPTION
**Pull request by knezmilos**
Fixes JIRA ticket DSD-1021

## This PR does the following:

- Adds capitalized, upper case, lower case text styles for `Heading` component

## How has this been tested?

Storybook

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Heading/Heading.tsx [hunk 1]
```diff
@@ -78,6 +93,24 @@ export const Heading = chakra(
...
+    if (isUppercase) {
+      textCase++;
+    }
+    if (isLowercase) {
+      textCase++;
+    }
+
+    if (textCase > 1) {
+      console.warn(
+        "NYPL Reservoir Text: Multiple text case props have been passed " +
```
Comment (bigfishdesign13) [inline src/components/Heading/Heading.tsx]: Typo.  Change to `NYPL Reservoir Heading: ...`

File: src/components/Heading/Heading.tsx [hunk 2]
```diff
@@ -52,6 +58,9 @@ export const Heading = chakra(
       id,
       level = "two",
       size,
+      isCapitalized,
```
Comment (bigfishdesign13) [inline src/components/Heading/Heading.tsx:59]: Please alphabetize the list of props.  It looks like one or two may have been out of order to begin with.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, this looke good.  There’s just one typo and one nit that need to be fixed.

File: src/components/Heading/Heading.tsx [hunk 3]
```diff
@@ -20,6 +20,12 @@ export interface HeadingProps {
   /** Optional size used to override the default styles of the semantic HTM
    * `<h>` elements */
   size?: HeadingSizes;
+  /** Optional prop used to show capitalized text */
```
Comment (EdwinGuzman) [inline src/components/Heading/Heading.tsx:17]: This list of props should also be alphabetized.

**Review by EdwinGuzman — APPROVED** [review]

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**